XBVC/emitters/templates/py_interface.jinja2

```diff
@@ -18,8 +21,45 @@ _ctypes_map = {
 {{msg.name.upper()}}_ID = {{msg.msg_id}}
 {% endfor %}
 
+MAX_PRECISION = 8
 
-def _encode_integer_to_bitvec(n, d_type):
+def _encode_float(n, max_precision=MAX_PRECISION):
+    work = n
+    whole = int(work)
+    work = work - whole
+    work = abs(work)
+
+    mul = 1
+    frac = 0
+    while max_precision and work > 0:
+        work *= 10
+        frac += int(work) * mul
+        work -= int(work)
+        max_precision -= 1
+        mul *= 10
+
+    return _encode_to_bitvec(whole, 's32') + _encode_to_bitvec(frac, 'u32')
+
+
+def _decode_float(lst):
+    whole, lst = _decode_from_bitvec(lst, 's32')
+    frac, lst = _decode_from_bitvec(lst, 'u32')
+    res_w = float(whole)
+    res_f = 0.0
+
+    mul = 10
+    while frac > 0:
+        res_f += (frac % 10) / mul
+        frac = frac // 10
+        mul *= 10
+
+    if res_w < 0:
+        res_f *= -1
+
+    return (res_w + res_f), lst
```

Comment:
Don't worry, I'll be documenting what these do in another PR (both a spec doc, as well as actual comments and tests here)